### ISSUE TITLE
Raise Z befor moving to Z_SAFE_HOMING

### ISSUE THREAD

**Issue by Gummibaer** [MarlinFirmware/Marlin#1815]
Please implement a Z raise for G28 Command
Befor Homing Z -> raise Z -> then moving to Z Safe

**Comment by thinkyhead** [comment]
@Gummibaer Currently it's enabled only for systems using auto-bed leveling, but I agree it should be available for all setups.

**Comment by TechMasterJoe** [comment]
testing that very thing got bored so printed a new control knob and did it at 100mm/s with silc3r 
I use Firmware Retraction, Volumetric and advance K =.008
lawl
![ndh9tu0ktvms18sg7rsl7dnditw33aayjkzox0dcl5q daqpdu-lylcymetrjgkjieaxib_on1wo_pn_zngpuck](https://cloud.githubusercontent.com/assets/11735057/6995986/51ca1ae2-db20-11e4-8093-450ff043abf7.jpg)
![-ctfktsq2m-ehzdyj0kmza8ohlki4rqaju3b1l1potg pomje1e83_bdvhxcaslq5dy2w5iayzmpvvnbxefi6dm](https://cloud.githubusercontent.com/assets/11735057/6995988/5b611b28-db20-11e4-8468-cb9c9762d200.jpg)
![7rwwom2uyr7nmzoittqqrqgj82isvmglrpjolcublpm wexndbdw4g5orszmrxivgvu_zhrqnjyeutupgq6azh8](https://cloud.githubusercontent.com/assets/11735057/6995989/5f1a7660-db20-11e4-97a7-35a2f46cbd15.jpg)

**Comment by Gummibaer** [comment]
@thinkyhead i have auto-bed leveling.
I need to send a G28 before the G29. by G28 crashed the hot end with a Screw.
I use corexy 
my solution in slic3r config
G1 Z10 ; raise Z 10mm
G28; homing all axes
G29; auto-bed leveling

**Comment by thinkyhead** [comment]
Have you tested the latest code? There was a fix in `G28` related to Z raise before homing within the last 20 hours or so. `G29` (and maybe `G29 E`) bed leveling are still wrong, so for the moment auto bed leveling needs to be compensated with a z homing offset, or go back to an earlier Marlin before Z raise was moved into `probe_pt()`… about 2 weeks ago.

**Comment by Gummibaer** [comment]
I have last from 04.04.2015 17:32.
this is a feuture request. Raise Z before homing

**Comment by thinkyhead** [comment]
What is your current setting for `Z_RAISE_BEFORE_HOMING`?

**Comment by Gummibaer** [comment]
`#define Z_RAISE_BEFORE_HOMING 10`
but it dosent work

Homingdirection

```
#define X_HOME_DIR -1
#define Y_HOME_DIR 1
#define Z_HOME_DIR -1
```

**Comment by thinkyhead** [comment]
@Gummibaer ~~Using a cartesian printer or a Delta?~~ I see, some of your home directions are -1, must be a cartesian!

**Comment by Gummibaer** [comment]
Yes its a cartesian. (the printer name is SparkCube)
Y is home direction to MAX

**Comment by jbrazio** [comment]
I believe this feature request was implemented with #3107.

**Comment by thinkyhead** [comment]
`MIN_Z_HEIGHT_FOR_HOMING` works for me!
